extensions/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/handler/WemoHandler.java

```diff
@@ -56,151 +49,113 @@
+public class WemoHandler extends BaseThingHandler implements UpnpIOParticipant {
 
     private final Logger logger = LoggerFactory.getLogger(WemoHandler.class);
 
-    public final static Set<ThingTypeUID> SUPPORTED_THING_TYPES = Sets.newHashSet(WEMO_SOCKET_TYPE_UID,
-            WEMO_INSIGHT_TYPE_UID, WEMO_LIGHTSWITCH_TYPE_UID, WEMO_MOTION_TYPE_UID, WEMO_SENSOR_TYPE_UID);
-    private static String getInsightParamsXML;
-    private static String getRequestXML;
-    private static String setRequestXML;
+    public final static Set<ThingTypeUID> SUPPORTED_THING_TYPES = Sets.newHashSet(THING_TYPE_SOCKET, THING_TYPE_INSIGHT,
+            THING_TYPE_LIGHTSWITCH, THING_TYPE_MOTION);
 
-    static {
-        try {
-            getInsightParamsXML = IOUtils.toString(WemoHandler.class
-                    .getResourceAsStream("/org/eclipse/smarthome/binding/wemo/internal/GetInsightParams.xml"));
-            getRequestXML = IOUtils.toString(WemoHandler.class
-                    .getResourceAsStream("/org/eclipse/smarthome/binding/wemo/internal/GetRequest.xml"));
-            setRequestXML = IOUtils.toString(WemoHandler.class
-                    .getResourceAsStream("/org/eclipse/smarthome/binding/wemo/internal/SetRequest.xml"));
-        } catch (Exception e) {
-            LoggerFactory.getLogger(WemoHandler.class).error("Cannot read XML files!", e);
-        }
-    }
+    private Map<String, String> stateMap = Collections.synchronizedMap(new HashMap<String, String>());
+
+    protected final static int SUBSCRIPTION_DURATION = 600;
+
+    private UpnpIOService service;
 
     /**
      * The default refresh interval in Seconds.
      */
-    private int refresh = 10;
+    private int DEFAULT_REFRESH_INTERVAL = 120;
 
     private ScheduledFuture<?> refreshJob;
 
-    public WemoHandler(Thing thing) {
+    private Runnable refreshRunnable = new Runnable() {
+
+        @Override
+        public void run() {
+            try {
+                updateWemoState();
+            } catch (Exception e) {
+                logger.debug("Exception during poll : {}", e);
+            }
+        }
+    };
+
+    public WemoHandler(Thing thing, UpnpIOService upnpIOService) {
+
         super(thing);
 
-        logger.debug("Create a WemoHandler for thing '{}'", getThing().getUID());
+        logger.debug("Creating a WemoHandler for thing '{}'", getThing().getUID());
+
+        if (upnpIOService != null) {
+            this.service = upnpIOService;
+        } else {
+            logger.debug("upnpIOService not set.");
+        }
     }
 
     @Override
     public void initialize() {
+
         Configuration configuration = getConfig();
 
-        if (configuration.get(UDN) != null) {
-            logger.debug("Initializing WemoHandler for UDN '{}'", configuration.get(UDN));
+        if (configuration.get("udn") != null) {
+            logger.debug("Initializing WemoHandler for UDN '{}'", configuration.get("udn"));
+            if (service.isRegistered(this)) {
```

Comment:
So if it is not registered at the time that the handler is initialized, you will never add a status listener then?